browser/brave_content_browser_client_browsertest.cc

```diff
@@ -483,6 +485,59 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest,
       << "No changes on the real URL";
 }
 
+IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, MixedContentForOnion) {
+  // Don't block the mock .onion requests.
+  tor::OnionLocationNavigationThrottle::BlockOnionRequestsOutsideTorForTesting(
+      false);
+
+  const GURL onion_url =
+      embedded_test_server()->GetURL("test.onion", "/onion.html");
+  const GURL onion_upgradable_url =
+      embedded_test_server()->GetURL("test.onion", "/onion_upgradable.html");
+  ASSERT_EQ("http", onion_url.scheme());
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), onion_upgradable_url));
+    EXPECT_TRUE(console_observer.messages().empty());
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    console_observer.SetPattern(
+        "Mixed Content: The page at 'http://test.onion*/onion.html' was "
+        "loaded over HTTPS, but requested an insecure image "
+        "'http://not_upgradable_to_https.com/image.jpg'. This content should "
+        "also be served over HTTPS.");
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), onion_url));
+    console_observer.Wait();
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    console_observer.SetPattern(
+        "Mixed Content: The page at 'http://test.onion*/onion.html' was "
+        "loaded over HTTPS, but requested an insecure resource "
+        "'http://example.com/'. This request has been blocked; the content "
+        "must be served over HTTPS.");
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('http://example.com')"));
+    console_observer.Wait();
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('https://example.com')"));
+    EXPECT_TRUE(console_observer.messages().empty() ||
+                console_observer.GetMessageAt(0).find(
+                    "has been blocked by CORS policy") != std::string::npos);
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('http://example.onion')"));
```

Comment:
also should change this to an onion site that sends `access-control-allow-origin: *`, ex: https://static01.nyt2sl3njavkomjxqe2e5nz6bsqv56yqbwkvhpmfn5jwh4pyccmjibad.onion/ads/tpc-check.html